nav2_experimental/nav2_rl/nav2_turtlebot3_rl/random_crawl_action.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) 2019 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os.path
+from ament_index_python.packages import get_package_share_directory
+from turtlebot3_env import TurtlebotEnv
+import numpy as np
+import rclpy
+import parameters
+
+from rclpy.node import Node
+from rclpy.action import ActionServer, CancelResponse, GoalResponse
+from time import sleep
+from keras.models import load_model
+
+import time
+
+from nav2_msgs.action import RandomCrawl
+from rclpy.executors import MultiThreadedExecutor
+from rclpy.callback_groups import ReentrantCallbackGroup
+import threading
+
+class RandomCrawlActionServer(Node):
+
+    def __init__(self):
+        super().__init__('action_server')
+        
+        self.env = TurtlebotEnv()
+        self.action_size = self.env.action_space()
+        print(self.action_size)
+        self.state = self.env.reset()
+        self.observation_space = len(self.state)
+        self.state = np.reshape(self.state, [1, self.observation_space])
+        pkg_share_directory = get_package_share_directory('nav2_turtlebot3_rl')
+        path = os.path.join(pkg_share_directory,"saved_models/random_crawl_waffle.h5")
+        self.model = load_model(path)
+        self.cancel_action = False
```

Comment:
You can check if the action was canceled through the goal handle: `goal_handle.is_cancel_requested`. Maybe use this instead and remove the variable.